### ISSUE TITLE
cabal update/list/info do not accept (the explicitly advertized) --no-require-sandbox flag

### ISSUE THREAD

**Issue by lspitzner** [haskell/cabal#2333]
~~~~
> cabal --version
cabal-install version 1.22.0.0
using version 1.22.0.0 of the Cabal library 
> cabal list
cabal: 'require-sandbox' is set to True, but no sandbox is present. Use
'--no-require-sandbox' if you want to override 'require-sandbox' temporarily.
> cabal list --no-require-sandbox
cabal: unrecognized 'list' option `--no-require-sandbox'
~~~~

similarly for update/info

i _would_ test this on latest HEAD if cabal would compile (attempts still pending)

**Comment by lspitzner** [comment]
the config option/flag `ignore-sandbox` is broken as well. config value seems to be ignored, and `--ignore-sandbox` is not accepted.
(btw require-sandbox=True + ignore-sandbox=True should always abort with error, right?)
(tested on 1.22.0.0)

**Comment by lspitzner** [comment]
this is mostly resolved by 121c9de9b3219b148ff633ebf242603e2f76917b (see #1889), but the `ignore-sandbox` config value still seems to have no effect (or i misinterpret its purpose); also, having an --ignore-sandbox flag might be useful (but low priority). will close this and open a new issue.
